packages/nx/src/plugins/js/project-graph/build-nodes/build-npm-package-nodes.ts

```diff
@@ -16,13 +17,16 @@ export function buildNpmPackageNodes(builder: ProjectGraphBuilder) {
     ...packageJson.devDependencies,
   };
   Object.keys(deps).forEach((d) => {
-    builder.addExternalNode({
-      type: 'npm',
-      name: `npm:${d}`,
-      data: {
-        version: deps[d],
-        packageName: d,
-      },
-    });
+    if (!builder.graph.externalNodes[`npm:${d}`]) {
+      builder.addExternalNode({
```

Comment:
I wonder if the addExernalNode call should throw if it would overwrite an existing node. Thoughts @FrozenPandaz ?

Reply:
If we implement check like we have for project node, e.g.:
```typescript
  addExternalNode(node: ProjectGraphExternalNode): void {
    // Check if project with the same name already exists
    if (this.graph.externalNodes[node.name]) {
      // Throw if another version already exists
      if (this.graph.externalNodes[node.name].data.version !== node.data.version) {
        throw new Error(
          `Multiple packages are named "${node.name}". One has version "${node.data.version
          }" and the other has version "${this.graph.externalNodes[node.name].data.version
          }". Please resolve the conflicting packages.`
        );
      }
    }
    this.graph.externalNodes[node.name] = node;
  }
```
this would break the existing `plugin` functionality. I will add a check to `addExternalNode` but I believe the condition should stay at `build-npm-package-nodes`